socs/agents/stm_thermo/agent.py

```diff
@@ -0,0 +1,147 @@
+
+        with self.lock.acquire_timeout(timeout=0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn(
+                    f'Could not start acq because {self.lock.job} is already running')
+                return False, 'Could not acquire lock.'
+
+            session.set_status('running')
+
+            self.take_data = True
+            session.data = {"fields": {}}
+            last_release = time.time()
+
+            while self.take_data:
+                if time.time() - last_release > LOCK_RELEASE_SEC:
+                    last_release = time.time()
+                    if not self.lock.release_and_acquire(timeout=LOCK_RELEASE_TIMEOUT):
+                        print(f'Re-acquire failed: {self.lock.job}')
+                        return False, 'Could not re-acquire lock.'
+
+                current_time = time.time()
+                data = {'timestamp': current_time, 'block_name': 'temps', 'data': {}}
+
+                for ch_num, dev in enumerate(self.devices):
+                    chan_string = f'Channel_{ch_num}'
+
+                    temp = dev.get_temp()
+                    data['data'][chan_string + '_T'] = temp
+
+                    if isinstance(dev, Max31856):
+                        cjtemp = dev.get_temp_ambient()
+                        data['data'][chan_string + '_T_cj'] = cjtemp
+                        field_dict = {chan_string: {'T': temp, 'T_cj': cjtemp}}
+                    else:
+                        field_dict = {chan_string: {'T': temp}}
+
+                    session.data['fields'].update(field_dict)
+
+                self.agent.publish_to_feed('temperatures', data)
+                session.data.update({'timestamp': current_time})
+                time.sleep(sleep_time)
+
+            self.agent.feeds['temperatures'].flush_buffer()
+
+        return True, 'Acquisition exited cleanly.'
+
+    def stop_acq(self, session, params=None):
+        """_stop_acq()
+
+        **Task** - Stop task associated with acq process.
+
+        """
+        if self.take_data:
+            self.take_data = False
+            return True, 'requested to stop taking data.'
+
+        return False, 'acq is not currently running.'
+
+
+def make_parser(parser=None):
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--paths_spinode', nargs='*', type=str, required=True,
+                        help="Path to the spi nodes.")
+
+    return parser
+
+
+def main(args=None):
+    '''Boot OCS agent'''
+    txaio.start_logging(level=os.environ.get('LOGLEVEL', 'info'))
+
+    parser = make_parser()
+    args = site_config.parse_args('StmThermometerAgent',
+                                  parser=parser,
+                                  args=args)
+
+    agent_inst, runner = ocs_agent.init_site_agent(args)
+    devices = [from_spi_node_path(path) for path in args.paths_spinode]
```

Comment:
You should move this into the agent itself, so that you can handle error cases within the agent.